sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -1832,4 +1832,18 @@ class AnalysisSuite extends AnalysisTest with Matchers {
     preemptedError.clear()
     assert(preemptedError.getErrorOpt().isEmpty)
   }
+
+  test("SPARK-49782: ResolveDataFrameDropColumns rule resolves complex UnresolvedAttribute") {
+    val function = UnresolvedFunction("trim", Seq(UnresolvedAttribute("i")), isDistinct = false)
+    val addF = Project(Seq(UnresolvedAttribute("i"), Alias(function, "f")()), testRelation5)
+    val dropF = DataFrameDropColumns(Seq(UnresolvedAttribute("f")), addF)
+    val dropRule = dropF.collectFirst {
```

Comment:
This looks unnecessary. We constructed `DataFrameDropColumns` explicitly and no need to check its existence.